src/displayapp/screens/StopWatch.cpp

```diff
@@ -123,97 +143,118 @@ void StopWatch::SetInterfaceStopped() {
   lv_obj_set_state(txtStopLap, LV_STATE_DISABLED);
 }
 
-void StopWatch::Reset() {
-  SetInterfaceStopped();
-  currentState = States::Init;
-  oldTimeElapsed = 0;
-  lapsDone = 0;
-}
-
-void StopWatch::Start() {
-  SetInterfaceRunning();
-  startTime = xTaskGetTickCount();
-  currentState = States::Running;
-  wakeLock.Lock();
+void StopWatch::RenderTime() {
+  TimeSeparated elapsedTime = ConvertTicksToTimeSegments(stopWatchController.GetElapsedTime());
+  renderedSeconds = elapsedTime.epochSecs;
+  if (renderedSeconds.IsUpdated()) {
+    SetHoursVisible(elapsedTime.hours != 0);
+    if (!hoursVisible) {
+      lv_label_set_text_fmt(time, "%02d:%02d", elapsedTime.mins, elapsedTime.secs);
+    } else {
+      lv_label_set_text_fmt(time, "%02d:%02d:%02d", elapsedTime.hours, elapsedTime.mins, elapsedTime.secs);
+    }
+  }
+  lv_label_set_text_fmt(msecTime, "%02d", elapsedTime.hundredths);
 }
 
-void StopWatch::Pause() {
-  SetInterfacePaused();
-  startTime = 0;
-  // Store the current time elapsed in cache
-  oldTimeElapsed = laps[lapsDone];
-  blinkTime = xTaskGetTickCount() + blinkInterval;
-  currentState = States::Halted;
-  wakeLock.Release();
+void StopWatch::RenderPause() {
+  const TickType_t currentTime = xTaskGetTickCount();
+  if (currentTime > blinkTime) {
```

Comment:
There's a tiny bug here when the tick count wraps around, but it wasn't introduced by this PR

Just before the count wraps around, the blink time in the future will always be smaller than the current time
Instead, tracking the last blink time and then doing `currentTime - lastBlinkTime > blinkInterval` is correct (since `currentTime - lastBlinkTime` is always the number of ticks since `lastBlinkTime`, regardless of wraparound)